appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroViewPager.kt

```diff
@@ -64,23 +65,19 @@ class AppIntroViewPager(context: Context, attrs: AttributeSet) : ViewPager(conte
     }
 
     fun goToPreviousSlide() {
-        try {
-            currentItem += if (!LayoutUtil.isRtl(context)) -1 else 1
-        } catch (e: Exception) {
-            LogHelper.e("AppIntro", "goToPreviousSlide: An error occurred while switching to the previous slide. Was isFirstSlide checked before the call?")
-        }
+        currentItem += if (!LayoutUtil.isRtl(context)) -1 else 1
     }
 
     fun isFirstSlide(size: Int): Boolean {
         return if (LayoutUtil.isRtl(context)) (currentItem - size + 1 == 0) else (currentItem == 0)
     }
 
-    fun getNextItem(size: Int) : Int {
-        return if (LayoutUtil.isRtl(context)) (size - currentItem) else currentItem + 1
+    fun getCurrentSlideNumber(size: Int): Int {
```

Comment:
The name is really confusing.
Can you explain the semantic of those two functions then:
`. currentItem`
`. getCurrentSlideNumber`

Reply:
@cortinico , We need these 2 methods because there is no rtl support in the current viewpager, resulting in some peculiar behaviour. Let me explain.

Case:1(Left to right layout)

Let's assume there are 4 slides,

currentitem: 0, 1, 2, 3
CurrentSlideNo.: 1, 2, 3, 4
Start ---> End

Case 2: (Right to left layout)
currentitem: 0, 1, 2, 3
CurrentSlideNo.: 4, 3, 2, 1
End<---Start

What I mean is that the order of viewpager items remain same in RTL mode, but the navigation direction changes, and since the permission request depends on the slide rather than the viewpager item, we need to correctly determine the slide number.(I found all of this after a lot of experimentation)